python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -145,22 +220,41 @@ def uniform_neighbor_sample(
         indices_t = numpy.int32
 
     if input_graph.renumbered:
-        start_list = input_graph.lookup_internal_vertex_id(start_list).compute()
+        start_list = input_graph.lookup_internal_vertex_id(start_list)
+    
+    start_list = start_list.rename(start_col_name).to_frame()
+    if batch_id_list is not None:
+        ddf = start_list.join(batch_id_list.rename(batch_col_name))
+    else:
+        ddf = start_list
+
+    if isinstance(ddf, cudf.DataFrame):
+        splits = cp.array_split(cp.arange(len(ddf)), len(Comms.get_workers()))
+        ddf = {
+            w: [ddf.iloc[splits[i]]]
+            for i,w in enumerate(Comms.get_workers())
+        }
+
+    else:
+        ddf = get_distributed_data(ddf)
+        wait(ddf)
+        ddf = ddf.worker_to_parts
 
     client = input_graph._client
 
     session_id = Comms.get_session_id()
-    start_list = cp.array_split(start_list.values, input_graph._npartitions)
 
     result = [
         client.submit(
             _call_plc_uniform_neighbor_sample,
             session_id,
             input_graph._plc_graph[w],
-            start_list[i],
+            ddf[w][0],
             fanout_vals,
             with_replacement,
             weight_t=weight_t,
+            with_edge_properties=with_edge_properties,
+            seed=seed+i, # ensure different seed per GPU
```

Comment:
How is this seed assigned and updated over the course of the execution of the sampling code?  It will be important that the seeds don't get reused across calls.  For example, if the caller simply increments the seed by 1, then each GPU will be reseeded in the next call with the neighbor GPUs sequence of random numbers - which is not going to be very random overall.

Part of the reason that the `cugraph_rng_state_t` object was added was to allow the state to be initialized once and maintained across calls - which will guarantee good uniqueness across multiple calls to the sampling algorithm.

Might be worth trying to think about how to propagate that design through into this layer as well.

Reply:
Without such a construct, you essentially rely on the caller understanding the mechanics of what is going on for the use of the random number generator and creating their own mechanism for providing unique seeds... which is an invitation for disaster.